save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/AwesomeBenchmarksDownloadController.kt

```diff
@@ -0,0 +1,103 @@
+import org.cqfn.save.preprocessor.utils.generateDirectory
+import org.eclipse.jgit.api.errors.GitAPIException
+import org.eclipse.jgit.api.errors.InvalidRemoteException
+import org.eclipse.jgit.api.errors.TransportException
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import com.akuleshov7.ktoml.file.TomlFileReader
+import org.cqfn.save.entities.benchmarks.BenchmarkEntity
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.reactive.function.BodyInserters
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.server.ResponseStatusException
+import reactor.core.publisher.Mono
+import kotlin.io.path.ExperimentalPathApi
+import kotlin.io.path.div
+
+/**
+ * A Spring controller for git project downloading
+ *
+ * @property configProperties config properties
+ */
+@OptIn(ExperimentalPathApi::class)
+@RestController
+class AwesomeBenchmarksDownloadController(
+        private val configProperties: ConfigProperties,
+) {
+    private val webClientBackend = WebClient.create(configProperties.backend)
+    private val tmpDir = generateDirectory(listOf(AWESOME_BENCHMARKS_URL), "${configProperties.repository}/awesome-benchmarks")
+
+    /**
+     * Controller to download standard test suites
+     *
+     * @return Empty response entity
+     */
+    @Suppress("TOO_LONG_FUNCTION", "TYPE_ALIAS")
+    @GetMapping("/download/awesome-benchmarks")
+    fun downloadAwesomeBenchmarks() =
+            Mono.just(ResponseEntity("Downloading awesome-benchmarks", HttpStatus.ACCEPTED))
+                    .doOnSuccess {
+                        log.info("Starting to download awesome-benchmarks to ${tmpDir.absolutePath}")
+                        cloneFromGit(gitDto, tmpDir)
+                        log.info("Awesome-benchmarks were downloaded to ${tmpDir.absolutePath}")
+                        processDirectoryAndCleanUp().subscribe()
+                        tmpDir.deleteRecursively()
+                    }
+                    .onErrorResume { exception ->
+                        tmpDir.deleteRecursively()
+                        when (exception) {
+                            is InvalidRemoteException,
+                            is TransportException,
+                            is GitAPIException -> log.warn("Error with git API while cloning ${gitDto.url} repository", exception)
+                            else -> log.warn("Cloning ${gitDto.url} repository failed", exception)
+                        }
+                        Mono.just(ResponseEntity("Downloading of awesome-benchmarks failed", HttpStatus.INTERNAL_SERVER_ERROR))
+                    }
+
+
+    private fun processDirectoryAndCleanUp(): Mono<Void> {
+        val resultList =
+                (tmpDir.toPath() / "benchmarks")
```

Comment:
"benchmarks" == some hardcoded name?

Reply:
benchmarks is a directory in  https://github.com/analysis-dev/awesome-benchmarks

Reply:
Won't it be better to extract it into properties to be able to easily change if needed?